go/vt/vtgate/planbuilder/testdata/select_cases.json

```diff
@@ -2278,70 +2278,60 @@
       "QueryType": "SELECT",
       "Original": "select (select col from user limit 1) as a from user join user_extra order by a",
       "Instructions": {
-        "OperatorType": "SimpleProjection",
```

Comment:
This plan was not working before these changes. Here we are ordering by the result of a subquery